vscode-car-plugin/src/main/java/org/wso2/maven/CARMojo.java

```diff
@@ -0,0 +1,263 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.maven;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.project.MavenProject;
+import org.wso2.maven.Model.ArtifactDependency;
+
+
+/**
+ * Goal which touches a timestamp file.
+ *
+ * @goal car
+ * @phase package
+ */
+@Mojo(name = "WSO2ESBDeployableArchive")
+public class CARMojo extends AbstractMojo {
+
+    /**
+     * Location archiveLocation folder
+     *
+     * @parameter expression="${project.basedir}"
+     */
+    private File projectBaseDir;
+
+    /**
+     * Location archiveLocation folder
+     *
+     * @parameter expression="${project.build.directory}"
+     */
+    private File archiveLocation;
+
+    /**
+     * finalName to use for the generated capp project if the user wants to override the default name
+     *
+     * @parameter
+     */
+    private String finalName;
+
+    /**
+     * @parameter default-value="${project}"
+     */
+    private MavenProject project;
+
+    /**
+     * A classifier for the build final name
+     *
+     * @parameter
+     */
+    private String classifier;
+
+    /**
+     * Read artifact.xml files and create .car file.
+     */
+    public void execute() throws MojoExecutionException {
+        appendLogs();
+        // Create CApp
+        try {
+            // Create directory to be compressed.
+            String archiveDirectory = getArchiveFile("").getAbsolutePath();
+            boolean createdArchiveDirectory = org.wso2.developerstudio.eclipse.utils.file.FileUtils.createDirectory(
+                    archiveDirectory);
+
+            if (createdArchiveDirectory) {
+                CAppHandler cAppHandler = new CAppHandler();
+                List<ArtifactDependency> dependencies = new ArrayList<>();
+
+                cAppHandler.processConfigArtifactXmlFile(projectBaseDir, archiveDirectory, dependencies);
+                cAppHandler.processRegistryResourceArtifactXmlFile(projectBaseDir, archiveDirectory, dependencies);
+                cAppHandler.createDependencyArtifactsXmlFile(archiveDirectory, dependencies, project);
+
+                File fileToZip = new File(archiveDirectory);
+                zipFolder(fileToZip.getPath(), getArchiveFile(".car").getPath());
+
+                recursiveDelete(fileToZip);
+
+            } else {
+                getLog().error("Could not create corresponding archive directory.");
+            }
+        } catch (Exception e) {
+            getLog().error(e);
```

Comment:
Shall we catch specific exception here, rather generic? 
Shouldn't we propergate the exception to the caller of execute() ?

Reply:
Created a custom exception class named "ArchiveException". Methods that are affected by IOException will catch the particular exception and will throw an ArchiveException to the caller execute().